lib/filterx/func-str-transform.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024 Axoflow
+ * Copyright (c) 2024 Attila Szakacs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx/func-str-transform.h"
+#include "filterx/object-string.h"
+#include "filterx/filterx-eval.h"
+
+static const gchar *
+_extract_str_arg(FilterXExpr *s, GPtrArray *args, gssize *len)
+{
+  if (args == NULL || args->len != 1)
+    {
+      filterx_simple_function_argument_error(s, "Requires exactly one argument", FALSE);
+      return NULL;
+    }
+
+  gsize inner_len;
+  FilterXObject *object = g_ptr_array_index(args, 0);
+  const gchar *str = filterx_string_get_value(object, &inner_len);
+  if (!str)
+    {
+      filterx_simple_function_argument_error(s, "Object must be string", FALSE);
+      return NULL;
+    }
+
+  *len = (gssize) MIN(inner_len, G_MAXINT64);
+  return str;
+}
+
+FilterXObject *
+filterx_simple_function_lower(FilterXExpr *s, GPtrArray *args)
+{
+  gssize len;
+  const gchar *str = _extract_str_arg(s, args, &len);
+  if (!str)
+    return NULL;
+
+  gchar *lower = g_ascii_strdown(str, len);
```

Comment:
I would suggest using [`utf8_strdown`](https://docs.gtk.org/glib/func.utf8_strdown.html) here, what do you think?

Reply:
Nice catch, fixed.

Reply:
I don't think we want UTF-8 here. Nothing ensures that a string is actually utf8.

Reply:
We do the same in the respective template function: https://github.com/alltilla/axosyslog/blob/1f82d7d4991c1e9e40d4060e0cec7830418cc5f8/modules/basicfuncs/str-funcs.c#L358
But I am okay to using ASCII.

Reply:
Ok then